wp_api/src/login.rs

```diff
@@ -60,15 +61,75 @@ pub struct WpApiDetails {
 
 #[uniffi::export]
 impl WpApiDetails {
+    /// Does the site have application passwords enabled?
     pub fn has_application_passwords_authentication_url(&self) -> bool {
         self.authentication
             .has_application_passwords_authentication_url()
     }
 
+    /// Returns the URL to be used in application password authentication.
+    ///
+    /// See the "Authorization Flow" section for details:
+    /// https://github.com/WordPress/wordpress-develop/blob/530493396b324f5bed518a494e2843e7fdb020f1/src/wp-includes/rest-api.php#L1099-L1119
     pub fn find_application_passwords_authentication_url(&self) -> Option<String> {
         self.authentication
             .find_application_passwords_authentication_url()
     }
+
+    /// Does the site URL (as defined by the site itself, not by user input) use HTTPS?
+    pub fn uses_https(&self) -> bool {
+        self.url.starts_with("https://")
+    }
+
+    /// Does the site use a plugin that disables application passwords?
+    pub fn has_application_password_blocking_plugin(&self) -> bool {
+        known_application_password_blocking_plugins()
+            .iter()
+            .any(|plugin| self.namespaces.contains(&plugin.namespace))
+    }
+
+    /// Returns a list of plugins that might be responsible for disabling application passwords.
+    pub fn application_password_blocking_plugins(
+        &self,
+    ) -> Vec<KnownApplicationPasswordBlockingPlugin> {
+        known_application_password_blocking_plugins()
+            .iter()
+            .filter(|plugin| self.namespaces.contains(&plugin.namespace))
+            .cloned()
+            .collect()
+    }
+
+    /// Returns the site URL (as defined by the site itself, not by user input) as a string.
+    pub fn site_url_string(&self) -> String {
+        self.url.clone()
+    }
+
+    /// Returns `true` if the site URL looks like a local development environment URL.
+    pub fn site_url_is_local_development_environment(&self) -> bool {
+        ParsedUrl::parse(self.url.as_str())
+            .is_ok_and(|parsed_url| is_local_dev_environment_url(&parsed_url))
+    }
+}
+
+pub fn known_application_password_blocking_plugins() -> Vec<KnownApplicationPasswordBlockingPlugin>
```

Comment:
I think we can make this an associated function for `KnownApplicationPasswordBlockingPlugin`. I've made this change in 5664111f6eaafeaa1a034ba374e546eed851635d also including the styling suggestion I mentioned. Please feel free to cherry-pick it if you agree with the suggestion.

Reply:
Accepted!